ag/validation.go

```diff
@@ -29,6 +31,11 @@ type idCleaner interface {
 // In addition, the interceptor also implements a cancel mechanism.
 func Interceptor(logger *zap.Logger) grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		methodName := strings.Split(info.FullMethod, "/")[2]
```

Comment:
I agree, but can you recommend a better solution?

Reply:
Not sure it will be safer, but this might be better option:
```go
methodName := info.FullPath[strings.LastIndex(info.FullPath, "/")+1:]
```
(it will either return -1 if the `/` is not found, which will cause the whole path to be returned in the `methodName`, or it will use only the last component of the `info.FullPath` input.

Reply:
That is, it won't panic because of index out of bounds.

Reply:
Yeah, this one looks more robust. Updated.